urwid/widget/scrollable.py

```diff
@@ -137,6 +137,17 @@ def get_first_visible_pos(self, size: tuple[int, int], focus: bool = False) -> i
     def get_visible_amount(self, size: tuple[int, int], focus: bool = False) -> int: ...
 
 
+def orig_iter(w: Widget) -> Iterator[Widget]:
+    visited = set((w,))
+    yield w
+    while hasattr(w, "original_widget"):
+        w = w.original_widget
+        if w in visited:
+            break
+        visited.add(w)
+        yield w
```

Comment:
http://urwid.org/reference/widget.html#urwid.WidgetDecoration.base_widget exists to get the "most" `.original_widget`

Reply:
Or are there decorations that are scrollable?

Reply:
> http://urwid.org/reference/widget.html#urwid.WidgetDecoration.base_widget exists to get the "most" `.original_widget`

Looking at the code, it is implemented like this generator was originally, i.e., vulnerable to infinite loops. So the fix in https://github.com/urwid/urwid/pull/879/commits/e5377cfa2cad87836e63bdfce257931e32b09662 should be applied also in `widget_decoration.py`, at least. 

I have no idea if there are scrollable decorations; this is my first time working with the urwid code. It seems like a silly thing to implement, but if someone did they might be hit with an extremely confusing bug if we only check the `base_widget`.

Reply:
> Or are there decorations that are scrollable?

`Scrollable` is a decoration adding scroll support to the widgets:
https://github.com/urwid/urwid/blob/ba5c1c07aef2b500f6d3f1a9e2b1453f5243904c/urwid/widget/scrollable.py#L140